src/simplicial-hott/04-extension-types.rzk.md

```diff
@@ -176,8 +176,8 @@ This is a literate `rzk` file:
         ( (t : χ) -> X t [ ψ t |-> f t ]))
   :=
     ( \ h -> (\ t -> h t , \ t -> h t) ,
-      ( ( \ fg t -> (second fg) t , \ h -> refl) ,
-        ( ( \ fg t -> (second fg) t , \ h -> refl))))
+      ( ( \ (_f, g) t -> g t , \ h -> refl) ,
```

Comment:
Can you explain the use of the underscore here and elsewhere?

Reply:
Single underscore (`_`) means a wildcard pattern (ignore the argument).
An identifier for an argument that starts with an underscore (e.g. `_f` or `_is-segal-A`) serves same purpose, but also helps readability, telling the reader a little bit about what the ignored argument represents.

This is a habit I transfer from programming in Haskell, I will gladly hear about @fredrik-bakke's experience in Agda.

Reply:
In Agda, the underscore character has multiple purposes.
1. It is used as a placeholder character for mixfix syntax. E.g. if you define a function `_!`, you can apply it by writing `x !`. Similarly, you could define the function `if_then_else_`. I think this is one of the big reasons why there are so few reserved characters and why spaces is so important in Agda.
2. It is used as a placeholder expression to tell Agda to infer a term.
3. It is used as a standalone token as a discard similar to how it functions in for instance C#. For example, if you want to define a constant function taking two arguments, you could write `\ _ _ -> x`.

I quite like having just `_` when discarding arguments as it looks more clean and doesn't clutter the code space with information that is irrelevant anyways. In `agda-unimath`, if an argument is not "irrelevant", but is not directly used in a definition, we tend to just give it a normal name I think. Other than that, I don't have a strong opinion either way and would be open to having a special naming convention for these. Maybe a suffix would look nicer than a prefix, but perhaps this defeats part of the purpose as it will be less familiar to people coming from Haskell or similar coding languages.